sdk/test-utils/recorder-v2/src/index.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as url from "url";
+import { DefaultHttpClient, WebResource, WebResourceLike } from "@azure/core-http";
```

Comment:
Starting a new thing with core-http feels kind of weird, given core-rest-pipeline is available. Or: do we need to have support for both and we are just starting with core-http?

Reply:
Need to support both, currently can be done only with core-http and for only some packages since not all packages allow passing httpClient as an option to the client.

Needs to be addressed - part of thee tasks at #15829

Reply:
I don’t see why we wouldn’t go with core-rest-pipeline as the first implementation though 🤔

Reply:
Started with core-http since that was simpler as httpClient can be provided as an option to the client.
Need to figure out core-rest-piepeline.

Reply:
At https://github.com/Azure/azure-sdk-for-js/pull/15826/commits/a27e3a209a4ae16412c273482bb6bf1c8948f4c4, both node and browser for both core-v1 and core-v2 work just fine.
A similar design from what I had done for the perf tests is being adopted.